http4k-realtime-core/src/testFixtures/kotlin/org/http4k/websocket/BlockingWebsocketClientContract.kt

```diff
@@ -31,16 +35,37 @@ abstract class BlockingWebsocketClientContract(
     }
 
     @Test
-    fun `send and receive in binary mode`() {
+    fun `send and receive in binary mode - memoryBody`() {
         val ws = websocket(Uri.of("ws://localhost:$port/bin"))
-        ws.send(WsMessage("hello".byteInputStream()))
+
+        val content = javaClass.classLoader.getResourceAsStream("org/http4k/websocket/sample_2k.png")!!.readBytes()
```

Comment:
Testing a string in binary mode isn't good enough.  We must ensure non-text binary isn't stripped out or coerced.